tgui/packages/tgui/interfaces/SecurityRecords.js

```diff
@@ -0,0 +1,435 @@
+  const {
+    id,
+    children,
+  } = properties;
+  return (
+    <Table.Cell>
+      <Button
+        color={sortId !== id && "transparent"}
+        width="100%"
+        onClick={() => {
+          if (sortId === id) {
+            setSortOrder(!sortOrder);
+          } else {
+            setSortId(id);
+            setSortOrder(true);
+          }
+        }}>
+        {children}
+        {sortId === id && (
+          <Icon
+            name={sortOrder ? "sort-up" : "sort-down"}
+            ml="0.25rem;"
+          />
+        )}
+      </Button>
+    </Table.Cell>
+  );
+};
+
+const SecurityRecordsActions = (properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    isPrinting,
+  } = data;
+  const [searchText, setSearchText] = useLocalState(context, "searchText", "");
+  return (
+    <Flex>
+      <FlexItem>
+        <Button
+          content="New Record"
+          icon="plus"
+          onClick={() => act('new_general')}
+        />
+        <Button
+          icon={isPrinting ? 'spinner' : 'print'}
+          disabled={isPrinting}
+          iconSpin={!!isPrinting}
+          content="Print Cell Log"
+          ml="0.25rem"
+          onClick={() => modalOpen(context, "print_cell_log")}
+        />
+      </FlexItem>
+      <FlexItem grow="1" ml="0.5rem">
+        <Input
+          placeholder="Search by Name, ID, Assignment, Fingerprint, Status"
+          width="100%"
+          onInput={(e, value) => setSearchText(value)}
+        />
+      </FlexItem>
+    </Flex>
+  );
+};
+
+const SecurityRecordsPageMaintenance = (properties, context) => {
+  const { act } = useBackend(context);
+  return (
+    <Box>
+      <Button
+        disabled
+        icon="download"
+        content="Backup to Disk"
+        tooltip="This feature is not available."
+        tooltipPosition="right"
+      /><br />
+      <Button
+        disabled
+        icon="upload"
+        content="Upload from Disk"
+        tooltip="This feature is not available."
+        tooltipPosition="right"
+        my="0.5rem"
+      /><br />
+      <Button.Confirm
+        icon="trash"
+        content="Delete All Security Records"
+        onClick={() => act('delete_security_all')}
+        mb="0.5rem"
+      /><br />
+      <Button.Confirm
+        icon="trash"
+        content="Delete All Cell Logs"
+        onClick={() => act('delete_cell_logs')}
+      />
+    </Box>
+  );
+};
+
+const SecurityRecordsPageView = (properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    isPrinting,
+    general,
+    security,
+  } = data;
+  if (!general || !general.fields) {
+    return (
+      <Box color="bad">
+        General records lost!
+      </Box>
+    );
+  }
+  return (
+    <Fragment>
+      <Section
+        title="General Data"
+        level={2}
+        mt="-6px"
+        buttons={
+          <Fragment>
+            <Button
+              icon={isPrinting ? 'spinner' : 'print'}
```

Comment:
Do spinners auto rotate? If not this might look a little off
![image](https://user-images.githubusercontent.com/25063394/94123877-3d70e580-fe4c-11ea-89d1-11c810f838e4.png)
The one missing pixel irks me

Reply:
```js
          icon={isPrinting ? 'spinner' : 'print'}
          disabled={isPrinting}
          iconSpin={!!isPrinting}
```

Yes